backup/barman-montagu

```diff
@@ -0,0 +1,290 @@
+            BARMAN_CONTAINER, barman.short_id))
+        return barman
+
+    pw = get_barman_passwords(password_group)
+    image_name = '{}/{}:{}'.format(DOCKER_REGISTRY, DOCKER_IMAGE, tag)
+    volume_mappings = {
+        BARMAN_VOLUMES['data']: {'bind': '/var/lib/barman'},
+        BARMAN_VOLUMES['logs']: {'bind': '/var/log/barman'},
+        BARMAN_VOLUMES['recover']: {'bind': '/recover'}
+    }
+    db = DATABASE_NAME.upper()
+    environment = {
+        'BARMAN_{}_PASS_BACKUP'.format(db): pw['barman'],
+        'BARMAN_{}_PASS_STREAM'.format(db): pw['streaming_barman']
+    }
+    extra_hosts = {'db': resolve_host_ip(host)}
+    if pull:
+        print("Pulling image...")
+        d.images.pull(image_name)
+    volumes = [volume_create(v, d) for v in BARMAN_VOLUMES.values()]
+    barman = d.containers.run(image_name, name=BARMAN_CONTAINER, detach=True,
+                              volumes=volume_mappings, environment=environment,
+                              extra_hosts=extra_hosts)
+    try:
+        exec_safely(barman, ["setup-barman"], True, True)
+    except:
+        if clean_on_error:
+            print("Cleaning up before exit")
+            barman.remove(force=True)
+            for v, newly_created in volumes:
+                if newly_created:
+                    v.remove()
+        raise
+    return barman
+
+
+def configure_cron():
+    print("Configuring cron")
+    if os.getuid() != 0:
+        raise Exception("Please rerun with sudo to install cron job")
+    command = "docker exec {} barman cron".format(BARMAN_CONTAINER)
+
+    if os.path.exists(PATH_CRON):
+        os.remove(PATH_CRON)
+    cron = crontab.CronTab(user=False)
+    job = cron.new(command=command, comment="Barman backup", user="root")
+    job.minute.every(1)
+    cron.write(PATH_CRON)
+    print("Wrote '{}' - remove this file to stop job".format(PATH_CRON))
+
+
+def vault_client():
+    vault_url = os.environ.get("VAULT_ADDR", DEFAULT_VAULT_ADDR)
+    vault_token = os.environ.get("VAULT_AUTH_GITHUB_TOKEN")
+    if not vault_token:
+        print("Please paste your vault GitHub access token: ", end="")
+        vault_token = input().strip()
+    vault = hvac.Client(url=vault_url)
+    print("Authenticating vault with GitHub")
+    vault.auth_github(vault_token)
+    return vault
+
+
+def get_barman_passwords(password_group):
+    users = ['barman', 'streaming_barman']
+
+    if password_group == "fake":
+        def get_password(user):
+            return user
+    else:
+        vault = vault_client()
+
+        def get_password(user):
+            path = "secret/database/{}/users/{}".format(password_group, user)
+            return vault.read(path)['data']['password']
+
+    return {x: get_password(x) for x in users}
+
+
+def get_barman_container(docker_client=docker.client.from_env(), error=True):
+    try:
+        return docker_client.containers.get(BARMAN_CONTAINER)
+    except docker.errors.NotFound:
+        pass
```

Comment:
Why not just put the error handling bit in the exception handler? This "return or pass, but then return" seems a little confusing.

Reply:
agreed